node/node.go

```diff
@@ -301,52 +301,65 @@ func (n *Node) stopInProc() {
 	}
 }
 
-// startIPC initializes and starts the IPC RPC endpoint.
-func (n *Node) startIPC(apis []rpc.API) error {
-	// Short circuit if the IPC endpoint isn't being exposed
-	if n.ipcEndpoint == "" {
-		return nil
-	}
+func StartIPCEndpoint(isClosedFn func() bool, ipcEndpoint string, apis []rpc.API) (net.Listener, *rpc.Server, error) {
```

Comment:
Could you please move `StartIPCEndpoint`, `StartHTTPEndpoint` (pls capitalized HTTP) and `StartWSEndpoint` (pls make one) into `rpc/endponts.go`? Looking at the code, it's purely RPC stuff, so it seems weird to have packages depend on `node`, which only want to do `rpc`.